src/Storages/PostgreSQL/PostgreSQLPoolWithFailover.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <Core/Types.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include "PostgreSQLConnectionPool.h"
+
+
+namespace postgres
+{
+
+class PoolWithFailover
```

Comment:
It's very similar to `mysqlxx::PoolWithFailover`, do you consider extracting common code to some generic abstraction?